### PR TITLE
Refactor shell scripts

### PR DESCRIPTION
**Pull request by stefaniuk**
## Description

This is the first part of the refactoring effort to ensure that the shell scripts:

- Follow a consistent convention,
- Pass the ShellCheck linting test (please run `make shellscript-lint-all`), and
- Work either with the pre-installed CLI tooling or run commands using Docker containers.

## Context

Information for a reviewer:

- The first two commits, [a4517db](https://github.com/nhs-england-tools/repository-template/pull/137/commits/a4517db0131bd0249bc6e4fef515bdd99de68d05) and [7b94f14](https://github.com/nhs-england-tools/repository-template/pull/137/commits/7b94f14921f83d0d99b7609da38c8d8b5b5829a8), implement stylistic and linting recommendations.
- The last two commits, [f8b9e84](https://github.com/nhs-england-tools/repository-template/pull/137/commits/f8b9e8408590a6ce3a47b5e96f9f2618e9c94bc9) and [22f7700](https://github.com/nhs-england-tools/repository-template/pull/137/commits/22f7700c362ede693d16071d6bb96cecb7f7200f), refactor how tools like Gitleaks and SonarScanner are executed.

## Type of changes

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by ballr73 — APPROVED** [review]

File: .github/actions/scan-secrets/action.yaml [hunk 1]
```diff
@@ -6,5 +6,5 @@ runs:
     - name: "Scan secrets"
       shell: bash
       run: |
-        export ALL_FILES=true # Do not change this line, as new patterns may be added or history may be rewritten
-        ./scripts/githooks/scan-secrets.sh
+        # Please, do not change it, as new patterns may be added or history may be rewritten.
```
Comment (regularfry) [inline .github/actions/scan-secrets/action.yaml:9]:
```suggestion
        # Please do not change this `check=whole-history` setting, as new patterns may be added or history may be rewritten.
```

File: scripts/githooks/check-file-format.sh [hunk 2]
```diff
@@ -45,9 +46,10 @@ image_version=2.7.1@sha256:dd3ca9ea50ef4518efe9be018d669ef9cf937f6bb5cfe2ef84ff2
 
 function main() {
 
-  cd $(git rev-parse --show-toplevel)
+  cd "$(git rev-parse --show-toplevel)"
 
-  is-arg-true "$dry_run" && dry_run_opt="--dry-run"
+  # shellcheck disable=SC2154
+  is-arg-true "${dry_run:-false}" && dry_run_opt="--dry-run"
```
Comment (regularfry) [inline scripts/githooks/check-file-format.sh:52]: Ooh, nice catch

File: scripts/githooks/check-file-format.sh [hunk 3]
```diff
@@ -89,9 +92,8 @@ function is-arg-true() {
 
 # ==============================================================================
 
+is-arg-true "${VERBOSE:-false}" && set -x
```
Comment (regularfry) [inline scripts/githooks/check-file-format.sh:95]: 👍

**Review by regularfry — CHANGES_REQUESTED** [review]
One slight clarification to a comment, otherwise this looks fine.  "Please" isn't usually followed by a comma, and I thought it worth pointing out the details of what it is that we're asking people not to change.

**Review by regularfry — APPROVED** [review]

**Pull request merged**